express-crud/server.ts

```diff
@@ -0,0 +1,148 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/grades', async (req, res, next) => {
+  try {
+    const sql = `
+      select * from "grades"
+    `;
+    const result = await db.query(sql);
+    const grades = result.rows;
+    res.status(200).json(grades);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/grades/:gradeId', async (req, res, next) => {
+  try {
+    const { gradeId } = req.params;
+    if (!Number.isInteger(+gradeId)) {
+      throw new ClientError(400, `Non-integer gradeId: ${gradeId}`);
+    }
+    const sql = `
+      select * from "grades"
+      where "gradeId" = $1;
+    `;
+    const params = [gradeId];
+    const result = await db.query(sql, params);
+    const grade = result.rows[0];
+    if (!grade) throw new ClientError(404, `grade ${gradeId} not found`);
+    res.status(200).json(grade);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/grades', async (req, res, next) => {
+  try {
+    const { name, course, score } = req.body;
+
+    if (!name) {
+      throw new ClientError(400, 'name is required');
+    }
+    if (!course) {
+      throw new ClientError(400, 'course is required');
+    }
+    if (!score) {
+      throw new ClientError(400, 'score is required');
+    }
+    if (+score < 0 || +score > 100) {
+      throw new ClientError(400, 'valid score is required');
+    }
+
+    const sql = `
+      insert into "grades" ("name", "course", "score")
+      values ($1, $2, $3)
+      returning *;
+    `;
+    const result = await db.query(sql, [name, course, score]);
+    const grade = result.rows[0];
+    if (!grade) {
```

Comment:
This will never happen on an insert. If there is a problem inserting, it will throw an exception. So you can remove this check.